expo/router
# feat: expo-head native module

# Motivation


- Make it possible to obtain iOS SEO in Expo using the same native resolution as Safari: [ref](https://developer.apple.com/library/archive/documentation/General/Conceptual/AppSearch/WebContent.html#//apple_ref/doc/uid/TP40016308-CH8-SW3).
- Spoken explanation: https://twitter.com/Baconbrix/status/1641850192839180315?s=20



# Execution

- Create an Expo Module for iOS which creates NSUserActivity objects based on the contents of the Head element.
- Disable in Expo Go.
- Still uses `react-helmet-async` on web.



# Test Plan

- Build `apps/sandbox` with `npx expo run:ios` and `npx expo run:android` (ensure it doesn't break)
- Run in Expo Go.
- Run in browser.
- `EXPO_USE_STATIC=1 npx expo-internal export -p web` -- ensure web still works.
